src/Microsoft.OData.Edm/Csdl/Serialization/EdmSchema.cs

```diff
@@ -48,7 +55,23 @@ public List<IEdmEntityContainer> EntityContainers
 
         public void AddSchemaElement(IEdmSchemaElement element)
         {
-            this.schemaElements.Add(element);
+            if (element.SchemaElementKind == EdmSchemaElementKind.Action ||
+                element.SchemaElementKind == EdmSchemaElementKind.Function)
+            {
+                IEdmOperation operation = (IEdmOperation)element;
+                IList<IEdmSchemaElement> operationList;
+                if (!this.operations.TryGetValue(operation.Name, out operationList))
+                {
+                    operationList = new List<IEdmSchemaElement>();
+                    this.operations[operation.Name] = operationList;
+                }
+
+                operationList.Add(operation);

```

Comment:
Is this not a breaking change, that operations now get added to SchemaOperations, rather than SchemaElements?

Reply:
Yes, but EdmSchema is an internal class and used internally

---
In reply to: [443058183](https://github.com/OData/odata.net/pull/1775#discussion_r443058183) [](ancestors = 443058183)